astroid/node_classes.py

```diff
@@ -599,8 +599,9 @@ def nearest(self, nodes):
                 "nodes %s and %s are not from the same module" % (self, node)
             )
             lineno = node.fromlineno
-            if node.fromlineno > mylineno:
-                break
+            if lineno > mylineno:
+                if nearest[1] > 0 and lineno-mylineno >= mylineno-nearest[1]:
```

Comment:
nearest[1] > 0 check here is necessary so that None is not returned because mylineno is closer to 0 compared to lineno.